lib/matplotlib/figure.py

```diff
@@ -375,6 +375,15 @@ def _repr_html_(self):
             from matplotlib.backends import backend_webagg
             return backend_webagg.ipython_inline_display(self)
 
+        if mpl.rcParams['backend'] == 'module://ipykernel.pylab.backend_inline':
```

Comment:
this should (likely?) check the canvas type (as for webagg), because rcParams["backend"] may change?

Reply:
That makes sense - should we have a figure output for all backends? And just choose the image format based on that backend?

Reply:
Instead of adding more coditional paths here we should make this defer to a `_repr_html_` on the canvas.

Reply:
@tacaswell Would adding a single `_repr_html_` to `backend_bases.FigureCanvaseBase` be enough?

Reply:
No, because for the interactive backends we do not want to replace them with a dead png.

Reply:
But this would only get triggered if a figure was the output of the last line of a cell in a notebook, which would (likely) only be done intentionally.